### PR TITLE
refactor(editors/subscriber...): Naming in GOOSE/SMV Subscriber plugins

### LINKED ISSUE #729: Naming in the subscription editor
**Is your feature request related to a problem? Please describe.**

I (am sorry,) didn't manage to look at this at the time the PR was raised. I think the subscriber wording could be clearer:

image

Any time we need 6-7 words to describe a radio button (and brackets!), the world seems to have become very complicated.

As a user of OpenSCD I want to not read a lot of words, and have descriptions of functionality be short and intuitive.

**Describe the solution you'd like**

I'd like to propose that the left radio button be: "Publisher | Subscriber" and the right radio button be "Subscriber | Publisher", perhaps something like this:

image

WDYT?

### LINKED ISSUE #958: Feedback on Later Binding SMV Plugin
**Is your feature request related to a problem? Please describe.**

This is some early, highly optional feedback on the excellent SMV Later Binding plugin. I did not manage to find time to comment during the review cycle, sorry.
We used this plugin (briefly) to configure some SMV subscriptions, and here is some feedback for enhancements at some stage.

**Describe the solution you'd like**

1. Initially, the person I tried out this extension on was confused about how to use it. I think perhaps some terminology changes would help.
    a. For the publisher on the left-hand side, I'd like to suggest a terminology change. Instead of "List of Sampled Values Controls with linked FCDAs" we could have (as per the other SMV plugin): "Sampled Values Publisher".
    b. For the subscriber on the left-hand side, I'd like to suggest we rename this from, "Selected Sampled Value data "smvcb0" / FCDA "AmpSv.instMag.i"" to: "Subscriber to". I'm not sure the control block name or the FCDA is useful if item 2 below is implemented. But if they are useful, I think including the IED name would be helpful. In many cases, merging units will have the same SMV control block name.

2. The selection on the LHS becomes lost when carrying out activity on the right-hand side. In other plugins (for instance, the Publisher plugin) we don't lose this, and it has a light green background (I think this is the `activated` property). This is valuable context for the user.

3. While the later binding plugin provides good functionality for the action of binding, it doesn't readily allow an overview of the configured ExtRefs which most tools provide, see the following screenshots from other applications which provide a tabular overview for each FCDA.

    a. AcSELerator Architect
        image

    b. Siemens DIGSI 5
        image

    c. We could do this through a separate plugin. Perhaps this is a useful overview for all the different editors. But if you are doing the mappings for a three-phase set of currents and voltages, for example, it is helpful to be able to see what is "already done and what is still to do".
    d. Another idea might be to allow clicking on the IED, and when this happens for all items in the SMV control block to be a synchronised list between the values on the left and right-hand screen, but this might not readily fit a mobile form factor.

### PR DESCRIPTION
**Pull request by danyill**
Closes #729 

This is in response to #729 and partially resolves #958 (slightly different terminology used than what was proposed there).

I know this is just wording and possibly doesn't seem that important :grey_question:. However, just a few words can make things more or less intuitive. I will be happy to receive a review, and if this is not helpful, I will be happy for it to be closed. :shrug: 

GOOSE/Sampled Value Control:

* Simplify the radio buttons to "Publisher | Subscriber" and "Subscriber | Publisher". They are too hard to parse and users must learn by doing in any case.
* Reword the headings for IED lists to be more explicit about what is happening in the two parts of the screen

Later Binding:

* If we are going to describe the source, we need to qualify it with not just the DA but the LDevice and LNode to be meaningful
* This is a lot of information, so an overflow should wrap to be responsive because this title can be pretty important
* Generally avoid technical language (like FCDAs and ExtRefs) and replace them with Messages and Inputs. I think this is better for users, although less precise.
* I decided not to include the IED name because I think applying the `activated` attribute to the filtered list will be adequate to provide this context (arguably removes a need for the title at all but in many ways looking up to more significant text is better than looking left)

The German is Google Translate and possibly no good; happy for it to be reverted or improved.

Here are some screenshots:

### GOOSE

**Publisher**

**Before**
image

**After**
image

Subscriber

**Before**
image

**After**
image

SMV is similar.

### SMV (Later Binding) 

**Before**
image

**After**
image

### CODE REVIEW

File: src/translations/de.ts [hunk 1]
```diff
@@ -346,53 +346,53 @@ export const de: Translations = {
...
+          title: 'Meldungen mit abgetasteten Werten',
+          noControlBlockFound: 'Keine Sampled Value Messages gefunden',
         },
       },
       GSEControl: {
         controlBlockList: {
-          title: '???',
-          noControlBlockFound: '???',
+          title: 'GOOSE-Meldungen',
+          noControlBlockFound: 'Keine GOOSE-Nachrichten gefunden',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:385]:
```suggestion
          noControlBlockFound: 'Keine GOOSEs gefunden',
```

**Review by JakobVogelsang — DISMISSED** [review]
Thanks for this very important contribution! There are some little details that need to be changed, but it is on a good track. Before fixing any of it I would like to ask for a rebase on the current main. As we worked on the Later binding GOOSE in parallel you will have merge conflicts otherwise

File: src/editors/subscription/smv-laterbinding/ext-ref-laterbinding-list.ts [hunk 2]
```diff
@@ -217,6 +225,8 @@ export class ExtRefLaterBindingList extends LitElement {
         'subscription.laterBinding.extRefList.SampledValueControl.title',
         {
           svcName: svcName ?? '-',
+          lDeviceName: lDeviceName ?? '-',
+          lNodeName: lNodeName ?? '-',
```
[the remark this answers is not in the document]
Comment (danyill) [reply src/editors/subscription/smv-laterbinding/ext-ref-laterbinding-list.ts:229]: Yes, it is even longer after the rebase because it includes the control block name. We could rely on the activatable property to provide context for the selection on the LHS? WDYT?

File: test/unit/__snapshots__/Setting.test.snap.js [hunk 3]
```diff
@@ -50,26 +50,17 @@ snapshots["SettingElement saves chosen .nsdoc file and looks like latest snapsho
       </mwc-switch>
     </mwc-formfield>
   </form>
-  <openscd-divider>
-  </openscd-divider>
+  <wizard-divider>
+  </wizard-divider>
   <section>
-    <div style="overflow: hidden;">
```
[the remark this answers is not in the document]
Comment (danyill) [reply test/unit/__snapshots__/Setting.test.snap.js:56]: This is to allow that thing which, above is "a bit too long" to provide context even on small form factors.

[the remark this answers is not in the document]
Comment (danyill) [reply test/unit/__snapshots__/Setting.test.snap.js:56]: Oh whoops, this is not meant to be here. Somehow when regenerating snapshots, items which are unchanged by my code still regenerate and occasionally I incorrectly commit them.

File: src/translations/de.ts [hunk 4]
```diff
@@ -346,53 +346,53 @@ export const de: Translations = {
     },
     goose: {
       publisherGoose: {
-        title: 'GOOSE-Publizierer',
-        subscriberTitle: 'Verbunden mit {{ selected }}',
+        title: 'GOOSE-Verlag',
```
[the remark this answers is not in the document]
Comment (danyill) [reply src/translations/de.ts:349]: Reverted

**Comment by danyill** [comment]
Thank you for your review. I have (I hope) done as requested. Just the outstanding issue about how to treat the title, I am interested in your thoughts and a further review. Particularly I hope to have taken your advice on the German translation but it is easy to make mistakes...

File: src/editors/subscription/smv-laterbinding/ext-ref-laterbinding-list.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply src/editors/subscription/smv-laterbinding/ext-ref-laterbinding-list.ts]: Yes, that is actually a suggestion from other contributors as well. So long we can permanently see the selected `FCDA` there is no need for a title here:

[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply src/editors/subscription/smv-laterbinding/ext-ref-laterbinding-list.ts]: But then I would add this feature in this PR directly and get rid of the title :)

**Comment by danyill** [comment]
I had a go at adding activatable.

The follow on from this is that in the GOOSE and Subscriber plugins prior to this PR the list selection gets lost when changing between editors.

Either we should retain a reference and restore it or we should make it deselected on first entry to the plugin (which is what I think I have done using `firstUpdated` but I am unsure if this is a good approach).

Note that the radio button in the GOOSE and Subscriber plugins has slightly odd behaviour. Toggling the radio button is one thing but when you click on it the selection is lost. I could not figure out exactly the cause of this (I was hoping to use the events triggered by the radio button to remove the selection references).

File: src/translations/de.ts [hunk 5]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
-        publisherTitle: 'Verbunden mit {{selected}}',
+      subscriber: {
+        iedListTitle: 'Sampled Value-Abonnenten',
+        publisherTitle: 'Sampled Value Messages abonniert bei {{ selected }}',
       },
       view: {
-        publisherView: 'Zeigt verbundene IED(s) der ausgewählten Sampled Value',
-        subscriberView:
-          'Zeigt verbundene Sampled Value(s) des ausgewählten IED',
+        publisherView: 'Herausgeber | Teilnehmer',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:371]:
```suggestion
        publisherView: 'Publisher | Subscriber',
```

File: src/translations/de.ts [hunk 6]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
+        subscriberView: 'Teilnehmer | Herausgeber',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:372]:
```suggestion
        subscriberView: 'Subscriber | Publkisher',
```

File: src/translations/de.ts [hunk 7]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
     smv: {
-      publisherSmv: {
+      publisher: {
         title: 'SampledValue-Publizierer',
         subscriberTitle: 'Verbunden mit {{ selected }}',
       },
-      subscriberSmv: {
-        publisherTitle: 'Verbunden mit {{selected}}',
+      subscriber: {
+        iedListTitle: 'Sampled Value-Abonnenten',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:367]:
```suggestion
        iedListTitle: 'Verbunden mit Sampled Values',
```

File: src/translations/de.ts [hunk 8]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
+        publisherTitle: 'Sampled Value Messages abonniert bei {{ selected }}',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:368]:
```suggestion
        publisherTitle: 'Sampled Value  verbunden mit {{ selected }}',
```

File: src/translations/de.ts [hunk 9]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
-      publisherGoose: {
+      publisher: {
         title: 'GOOSE-Publizierer',
         subscriberTitle: 'Verbunden mit {{ selected }}',
       },
-      subscriberGoose: {
-        publisherTitle: 'GOOSE(s) verbunden mit {{selected}}',
+      subscriber: {
+        iedListTitle: 'GOOSE-Abonnenten',
+        publisherTitle: 'GOOSE Messages abonniert bei {{ selected }}',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:354]:
```suggestion
        publisherTitle: 'GOOSE Nachricht verbunden mit {{ selected }}',
```

File: src/translations/de.ts [hunk 10]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
     goose: {
-      publisherGoose: {
+      publisher: {
         title: 'GOOSE-Publizierer',
         subscriberTitle: 'Verbunden mit {{ selected }}',
       },
-      subscriberGoose: {
-        publisherTitle: 'GOOSE(s) verbunden mit {{selected}}',
+      subscriber: {
+        iedListTitle: 'GOOSE-Abonnenten',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:353]:
```suggestion
        iedListTitle: 'Verbunden mit GOOSE Meldungen',
```

File: src/translations/de.ts [hunk 11]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
-          title: '???',
-          noControlBlockFound: '???',
+          title: 'GOOSE-Meldungen',
+          noControlBlockFound: 'Keine GOOSEs gefunden',
         },
       },
       extRefList: {
-        title:
-          'Ausgewählter {{ controlTag }} Datenpunkt "{{ svcName }}" / "{{ fcdaName }}"',
+        title: 'Für Ausgewählte Daten-Attribut Verfügbare Eingaben',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:389]:
```suggestion
        title: 'Für Ausgewählte Daten-Attribut Verfügbare Verbindung',
```

File: src/translations/de.ts [hunk 12]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
         noSelection: 'Kein Daten-Attribut ausgewählt',
-        noSubscribedExtRefs: 'Keine verknüpften externen Referenzen',
-        noAvailableExtRefs:
-          'Keine externen externen Referenzen vom Type "Later Binding"',
+        noSubscribedExtRefs: 'Keine abonnierten Eingänge',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:391]:
```suggestion
        noSubscribedExtRefs: 'Keine bestehenden Verbindungen',
```

File: src/translations/de.ts [hunk 13]
```diff
@@ -345,56 +345,55 @@ export const de: Translations = {
...
+        noAvailableExtRefs: 'Keine verfügbaren Eingaben zum Abonnieren',
```
Comment (JakobVogelsang) [inline src/translations/de.ts:392]:
```suggestion
        noAvailableExtRefs: 'Keine verfügbaren Eingänge vorhanden',
```

**Review by JakobVogelsang — CHANGES_REQUESTED** [review]
In general LGTM. I fixed some German translations, and you need to merge main, and I think update the snapshots as @juancho0202 was working on the same code. 

Nice @danyill very clean and nice changes

**Comment by danyill** [comment]
Thank you for this review :bow:, for the German translations and for accepting this slightly fiddly PR. I have made these changes and merged `main` to this branch, no snapshots needed updating.

**Review by JakobVogelsang — APPROVED** [review]
LGTM

**Comment by JakobVogelsang** [comment]
I am sorry but I have to trigger you again @danyill. @juancho0202 has pushed last changes in the Later binding plugin and that is conflicting you work. Could you do another merge? We have nothing conflicting in the pipeling at the moment so this will not be wasted time :D

**Comment by juancho0202** [comment]
Sorry @danyill 😅

**Comment by danyill** [comment]
No problem. This time we courageously use the merge button though our :heart: trembles.
